### PR TITLE
Move RestClient to SignalR.Common

### PR DESCRIPTION
**Pull request by terencefan**

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Common/Utilities/RestApiProvider.cs [hunk 1]
```diff
@@ -3,14 +3,14 @@
 
 using System;
 
-namespace Microsoft.Azure.SignalR.Management
+namespace Microsoft.Azure.SignalR
 {
     internal class RestApiProvider
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Common/Utilities/RestApiProvider.cs:8]: leave this to management project?

File: build/dependencies.props [hunk 2]
```diff
@@ -6,8 +6,11 @@
     <!-- Azure ASP.NET Core SignalR -->
     <MicrosoftAspNetCoreHttpConnectionsClientPackageVersion>1.0.0</MicrosoftAspNetCoreHttpConnectionsClientPackageVersion>
     <MicrosoftAspNetCoreHttpConnectionsClientPackage3_0Version>3.0.0</MicrosoftAspNetCoreHttpConnectionsClientPackage3_0Version>
-    <MicrosoftExtensionsLoggingAbstractionsPackageVersion>1.0.0</MicrosoftExtensionsLoggingAbstractionsPackageVersion>
+    <MicrosoftExtensionsLoggingAbstractionsPackageVersion>2.1.0</MicrosoftExtensionsLoggingAbstractionsPackageVersion>
```
Comment (vicancy) [inline build/dependencies.props:9]: don't update the version unless neccessory

**Comment by wanlwanl** [comment]
Why moving it to common? Anywhere uses the rest client except for Management SDK?

**Comment by terencefan** [comment]
@wanlwanl Our SDK may call `GetAccessKey` API to create a temporary AccessKey in the future.

File: build/dependencies.props (hunk 2 above)
Comment (terencefan) [reply build/dependencies.props:9]: `Extension.Http=2.1.0` has been introduced, who depend on the 2.1.0 version of Logging.

So we have to upgrade it anyway.

**Pull request merged**